src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/DomainEventService.kt

```diff
@@ -249,4 +345,20 @@ class DomainEventService(
     ApplicationWithdrawnEnvelope::class.java -> DomainEventType.APPROVED_PREMISES_APPLICATION_WITHDRAWN
     else -> throw RuntimeException("Unrecognised domain event type: ${type.name}")
   }
+
+  final fun retryTemplate(): RetryTemplate {
+    val retryTemplate = RetryTemplate()
+    val backOffPolicy = FixedBackOffPolicy()
+    backOffPolicy.backOffPeriod = BACK_OFF_PERIOD
+    retryTemplate.setBackOffPolicy(backOffPolicy)
+    val retryPolicy = SimpleRetryPolicy()
+    retryPolicy.maxAttempts = MAX_ATTEMPTS_RETRY
+    retryTemplate.setRetryPolicy(retryPolicy)
+    return retryTemplate
+  }
+
+  companion object {
+    const val MAX_ATTEMPTS_RETRY = 4
```

Comment:
What do we think about 4 here? I think if these requests ever fail something serious has gone wrong that may persist for 12h or 24h. I worry about us releasing a change that causes failures and by the time we notice and fix it the queue has given up. How about the equivalent of 30 days?

Reply:
Keep trying every 1000 milliseconds for 30 days?

Reply:
I'd assume an increasing backoff though judging our caching I presume we'd have to figure that bit out?

Every second is too often 😅 

Perhaps 4 at 1 minute intervals, 4 at 10 minutes intervals and then however many an hour to give us a big window for it to automatically recover.

Reply:
OK I'll look into how we do this with retry template or if it's even possible.

Reply:
@tahb I've used the ExponentialBackOffPolicy with it's default multiplier of 2. So we start off with 1 second then 2 second then 4 second etc until we get to ten minutes. We stick with ten minutes for the next seven days. Discussed with Stu that seven days should be adequate.